src/main/java/blusunrize/immersiveengineering/api/tool/ToolboxHandler.java

```diff
@@ -0,0 +1,68 @@
+package blusunrize.immersiveengineering.api.tool;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.BiPredicate;
+import java.util.function.Predicate;
+
+import blusunrize.immersiveengineering.api.energy.wires.IImmersiveConnectable;
+import blusunrize.immersiveengineering.api.energy.wires.IWireCoil;
+import net.minecraft.block.Block;
+import net.minecraft.block.state.IBlockState;
+import net.minecraft.item.ItemFood;
+import net.minecraft.item.ItemStack;
+import net.minecraft.item.ItemTool;
+import net.minecraft.world.World;
+
+public class ToolboxHandler {
+	private static final List<Predicate<ItemStack>> tools = new ArrayList<>();
+	private static final List<Predicate<ItemStack>> foods = new ArrayList<>();
+	private static final List<BiPredicate<ItemStack, World>> wiring = new ArrayList<>();
+	static {
+		tools.add((s)->(s.getItem() instanceof ITool?((ITool)s.getItem()).isTool(s):false));
+		tools.add((s)->(s.getItem() instanceof ItemTool));
+		foods.add((s)->(s.getItem() instanceof ItemFood));
```

Comment:
Both yes! I heard you say that on stream and thought that it would be a real nice use case for lambdas. It's the first time I have actually used them, but they are quite nice for this kind of thing.